core/src/services/hdfs/backend.rs

```diff
@@ -387,10 +391,13 @@ impl Access for HdfsBackend {
 
     async fn rename(&self, from: &str, to: &str, _args: OpRename) -> Result<RpRename> {
```

Comment:
I didn't read the details of the implementation. At a glance, this looks similar to `blocking_rename`. Maybe we can build `HdfsCore::hdfs_rename`. This will benefit:

- `rename`
- `blocking_rename`